File: Shoppy/src/app/features/product/models/Product.ts
import { Category } from "./Category";

export interface Product {
    id: number;
    name: string;
    description: string;
    price: number;
    stockQuantity: number;
    categoryId: number;
    imageUrl: string;
    createdDate: string;
    updatedDate?: string;
    category: Category;
}

File: ProjectSaleStore/ProjectSaleStore/Models/Product.cs

namespace SaleStoreDecor.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int StockQuantity { get; set; } = 0;
        public int CategoryId { get; set; }
        public string ImageUrl { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.Now;
        public DateTime? UpdatedDate { get; set; }

        public Category Category { get; set; }
        public ICollection<OrderItem> OrderItems { get; set; }
        public ICollection<CartItem> CartItems { get; set; }
        public ICollection<Feedback> Feedbacks { get; set; }
    }
}
